src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +565,222 @@ subroutine DustEmission (bounds, &
+               flx_mss_vrt_dst_ttl(p) = flx_mss_vrt_dst_ttl(p) * lnd_frc_mbl(p) * C_tune * liqfrac
+               !########################################################################################
+            end if
+
+            !############## Danny M. Leung added the intermittency calculation #################################
+            ! subsection for intermittency factor calculation (only used by Leung's scheme, not Zender's scheme)
+            ! 2 Dec 2021: assume no buoyancy contribution to the wind fluctuation (u_sd_slt), so no obul(p) is needed. It is shown to be important for the wind fluctuations contribute little to the intermittency factor. We might add this back in the future revisions.
+
+            ! mean lowpass-filtered wind speed at 0.1 m saltation height (assuming aerodynamic roughness length = 1e-4 m globally for ease; also assuming neutral condition)
+            u_mean_slt(p) = (wnd_frc_slt/k) * log(0.1_r8 / 1e-4_r8)  ! translating from ustar (velocity scale) to actual wind
+
+            ! sd of lowpass-filtered wind speed
+            !if (obul(p)==0) then
+            !   zetaobu = 0
+            !else 
+               !zetaobu = zii(p) / obul(p)   ! For now zii is a constant of 1000 m in CLM -dml, 24 Aug 2021
+            !   zetaobu = 1000_r8 / obul(p)   ! For now zii is a constant of 1000 m in CLM -dml, 24 Aug 2021
+            !end if
+            !stblty(p) = zetaobu    ! zetaobu get outputted as the Obukhov stability parameter
+            stblty(p) = 0   ! -dmleung 2 Dec 2021: use stability = 0 for now, assuming no buoyancy contribution. Might uncomment the above lines in future revisions.
+            if ((12_r8 - 0.5_r8 * stblty(p)) .GE. 0.001_r8) then ! should have used 0 theoretically; used 0.001 here to avoid undefined values
+               u_sd_slt(p) = wnd_frc_slt * (12_r8 - 0.5_r8 * stblty(p))**0.333_r8
+            else
+               u_sd_slt(p) = 0.001_r8   ! should have used 0 theoretically; used 0.001 here to avoid undefined values
             end if
 
-            ! the following comes from subr. flx_mss_vrt_dst_ttl_MaB95_get
-            ! purpose: diagnose total vertical mass flux of dust from vertically
-            !          integrated streamwise mass flux
+            ! threshold velocities
+            ! Here wnd_frc_thr_slt is the fluid threshold; wnd_frc_thr_dry(p) is the dry fluid threshold; B_it*wnd_frc_thr_dry(p) is the impact threshold
+            ! fluid threshold wind at 0.1 m saltation height
+            u_fld_thr(p) = (wnd_frc_thr_slt/k) * log(0.1_r8 / 1e-4_r8)
+            ! impact threshold wind at 0.1 m saltation height
+            u_impct_thr(p) = (wnd_frc_thr_slt_it/k) * log(0.1_r8 / 1e-4_r8)  ! to avoid model error
+
+            ! threshold crossing rate
+            numer = (u_fld_thr(p)**2_r8 - u_impct_thr(p)**2_r8 - 2_r8 * u_mean_slt(p) * (u_fld_thr(p) - u_impct_thr(p)))
```

Comment:
I think we should change the 2_r8 into 2.0_r8 just for readability.

Reply:
Done.